Tribler/Core/Modules/freespace_checker.py

```diff
@@ -0,0 +1,22 @@
+from twisted.internet.task import LoopingCall
+from Tribler.Core.osutils import get_free_space
+from Tribler.Core.simpledefs import NTFY_FREE_SPACE, NTFY_INSERT
+from Tribler.Main.globals import DefaultDownloadStartupConfig
+from Tribler.dispersy.taskmanager import TaskManager
+
+
+class FreeSpaceChecker(TaskManager):
+
+    def __init__(self, session):
+        super(FreeSpaceChecker, self).__init__()
+
+        self.session = session
+        self.check_path = DefaultDownloadStartupConfig.getInstance().get_dest_dir()
+
+        self.free_space_check_task = self.register_task("check free disk space ", LoopingCall(self.check_free_space))
```

Comment:
I would have a start() and stop() method on all these components so that instancing them doesn't mean they will be active immediately.